parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java

```diff
@@ -300,6 +301,8 @@ public void close() throws IOException {
     } catch (InterruptedException e) {
       throw new IOException(e);
     }
+    // release after the writer closes in case it is used for a last flush
+    codecFactory.release();
```

Comment:
should this be in a finally block to release in the case where an exception occurs?